fuel-poa-coordinator/src/service.rs

```diff
@@ -0,0 +1,326 @@
+use crate::{
+    deadline_clock::{
+        DeadlineClock,
+        OnConflict,
+    },
+    Config,
+    Trigger,
+};
+use anyhow::Context;
+use fuel_core_interfaces::{
+    block_importer::ImportBlockBroadcast,
+    block_producer::BlockProducer,
+    common::{
+        fuel_vm::prelude::InterpreterStorage,
+        prelude::Word,
+    },
+    model::{
+        BlockHeight,
+        FuelBlock,
+    },
+    txpool::{
+        Sender as TxPoolSender,
+        TxStatus,
+        TxStatusBroadcast,
+    },
+};
+use parking_lot::Mutex;
+use std::sync::Arc;
+use tokio::{
+    sync::{
+        broadcast,
+        mpsc,
+    },
+    task::JoinHandle,
+    time::Instant,
+};
+use tracing::{
+    error,
+    warn,
+};
+
+pub struct RunningService {
+    join: JoinHandle<()>,
+    stop: mpsc::Sender<()>,
+}
+
+pub struct Service {
+    running: Mutex<Option<RunningService>>,
+    config: Config,
+}
+
+impl Service {
+    pub fn new(config: &Config) -> Self {
+        Self {
+            running: Mutex::new(None),
+            config: config.clone(),
+        }
+    }
+
+    pub async fn start<S>(
+        &self,
+        txpool_broadcast: broadcast::Receiver<TxStatusBroadcast>,
+        txpool_sender: TxPoolSender,
+        import_block_events_tx: broadcast::Sender<ImportBlockBroadcast>,
+        block_producer: Arc<dyn BlockProducer>,
+        db: S,
+    ) where
+        S: InterpreterStorage + Send + Clone + 'static,
```

Comment:
See comment above: https://github.com/FuelLabs/fuel-core/pull/599/files#r991677314

It's better to create traits with similar getters, than to reuse traits with many other useless things (from a PoA perspective) that make mocking difficult here.

Reply:
Done